eZ/Publish/Core/FieldType/RichText/InternalLinkValidator.php

```diff
@@ -54,10 +90,35 @@ public function validate($scheme, $id)
                 default:
                     throw new InvalidArgumentException($scheme, "Given scheme '{$scheme}' is not supported.");
             }
+        } catch (UnauthorizedException $e) {
+            // Editor can link to content/location even if he doesn’t have permissions
```

Comment:
should we use spi here instead to avoid this? would be a bit faster /cc @alongosz

Reply:
@adamwojs so, have you tried to use SPI here? Suggestion looks good (it's fine for Core).